lib/yaml_permitted_classes.rb

```diff
@@ -0,0 +1,21 @@
+class YamlPermittedClasses
+  DEFAULT_PERMITTED_CLASSES = [Object, Range, Regexp, Symbol, Date, Time, DateTime, ActiveSupport::Duration, ActiveSupport::HashWithIndifferentAccess, ActiveSupport::TimeWithZone, ActiveSupport::TimeZone]
+  def self.app_yaml_permitted_classes
+    @app_yaml_permitted_classes ||= DEFAULT_PERMITTED_CLASSES + [MiqExpression, MiqReport, Ruport::Data::Table, Ruport::Data::Record, User, ConfigurationScript, ContainerImage, ContainerTemplate, OrchestrationTemplate, ManageIQ::Providers::Vmware::InfraManager, ManageIQ::Providers::InfraManager::Vm]
```

Comment:
This is going to be pretty annoying if we basically eager load all of our models we permit to be yaml loaded from serialized column.

Reply:
`DEFAULT_PERMITTED_CLASSES` contains libraries so that looks ok.
I am concerned about `@app_yml_permitted_classes` since that has classes from `manageiq/app/models` which can be dynamically reloaded.

Is there a rails hook we can use to clear the `@app_` cache variable?

Reply:
Agreed.  There is likely a reload hook we can wrap this and clear it.

I'm not in love with the current way to specify classes that can be loaded via YAML specifically because of the point you're using.  It's taking actual class objects, not a symbol/string of the class so we need to eager load the classes and as you said, deal with the class being reloaded.